src/Utils/BlockTemplateUtils.php

```diff
@@ -74,6 +74,143 @@ public static function gutenberg_inject_theme_attribute_in_content( $template_co
+		);
+		foreach ( $themes as $theme_slug => $theme_dir ) {
+			$file_path = $theme_dir . '/' . $template_base_paths[ $template_type ] . '/' . $slug . '.html';
+			if ( file_exists( $file_path ) ) {
+				$new_template_item = array(
+					'slug'  => $slug,
+					'path'  => $file_path,
+					'theme' => $theme_slug,
+					'type'  => $template_type,
+				);
+
+				if ( 'wp_template_part' === $template_type ) {
+					return self::gutenberg_add_template_part_area_info( $new_template_item );
+				}
+
+				if ( 'wp_template' === $template_type ) {
+					return self::gutenberg_add_template_info( $new_template_item );
+				}
+
+				return $new_template_item;
+			}
+		}
+
+		return null;
+	}
+
+	/**
+	 * Attempts to add the template part's area information to the input template.
+	 *
+	 * @param array $template_info Template to add information to (requires 'type' and 'slug' fields).
+	 *
+	 * @return array Template.
+	 */
+	public static function gutenberg_add_template_part_area_info( $template_info ) {
+		if ( \WP_Theme_JSON_Resolver_Gutenberg::theme_has_support() ) {
+			$theme_data = \WP_Theme_JSON_Resolver_Gutenberg::get_theme_data()->get_template_parts();
+		}
+
+		if ( isset( $theme_data[ $template_info['slug'] ]['area'] ) ) {
+			$template_info['title'] = $theme_data[ $template_info['slug'] ]['title'];
+			$template_info['area']  = gutenberg_filter_template_part_area( $theme_data[ $template_info['slug'] ]['area'] );
+		} else {
+			$template_info['area'] = WP_TEMPLATE_PART_AREA_UNCATEGORIZED;
+		}
+
+		return $template_info;
+	}
+
+	/**
+	 * Build a unified template object based a post Object.
+	 *
+	 * @param \WP_Post $post Template post.
+	 *
+	 * @return \WP_Block_Template|\WP_Error Template.
+	 */
+	public static function gutenberg_build_template_result_from_post( $post ) {
+		$default_template_types = gutenberg_get_default_template_types();
+		$terms                  = get_the_terms( $post, 'wp_theme' );
+
+		if ( is_wp_error( $terms ) ) {
+			return $terms;
+		}
+
+		if ( ! $terms ) {
+			return new \WP_Error( 'template_missing_theme', __( 'No theme is defined for this template.', 'woo-gutenberg-products-block' ) );
+		}
+
+		$theme          = $terms[0]->name;
+		$has_theme_file = true;
+
+		$template                 = new \WP_Block_Template();
+		$template->wp_id          = $post->ID;
+		$template->id             = $theme . '//' . $post->post_name;
+		$template->theme          = $theme;
+		$template->content        = $post->post_content;
+		$template->slug           = $post->post_name;
+		$template->source         = 'custom';
+		$template->type           = $post->post_type;
+		$template->description    = $post->post_excerpt;
+		$template->title          = $post->post_title;
+		$template->status         = $post->post_status;
+		$template->has_theme_file = $has_theme_file;
+		$template->is_custom      = true;
+
+		if ( 'wp_template' === $post->post_type && isset( $default_template_types[ $template->slug ] ) ) {
```

Comment:
I think we can delete this as it will never resolve as the Woo templates don't exist in `$default_template_types`